lib/host.js

```diff
@@ -305,7 +335,47 @@ HostConnectionPool.prototype.forceShutdown = function () {
   });
 };
 
+/**
+ * Closes all open connections and schedules reconnection
+ * @param {Number} delay
+ */
+HostConnectionPool.prototype.scheduleReconnection = function (delay) {
+  this._forceShutdown();
+  this.reconnectionTimeout = setTimeout(this._attemptReconnection.bind(this), delay);
+};
+
+/**
+ * Prevents reconnection timeout from triggering
+ */
+HostConnectionPool.prototype.clearReconnection = function () {
+  if (!this.reconnectionTimeout) {
+    return;
+  }
+  clearTimeout(this.reconnectionTimeout);
+  this.reconnectionTimeout = null;
+};
+
+HostConnectionPool.prototype._attemptReconnection = function () {
+  this.reconnectionTimeout = null;
+  var c = this._createConnection();
+  var self = this;
+  c.open(function (err) {
+    if (err) {
+      self.log('info', util.format('Reconnection attempt to host %s failed', self.address), err);
+      self.host.setDown();
+      return c.close();
+    }
+    self.host.setUp();
```

Comment:
Initially in `_maybeCreatePool` we could choose to invoke the callback immediately if the pool is being created and at least 1 core connection exists we invoke the callback, that way `borrowConnection` could in theory proceed granted the host is up and there was a successful reconnection.

```javascript
   if (!this.creating && this.connections.length >= this.coreConnectionsLength) {
     return callback();	
   }
```